src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -220,6 +231,23 @@ func (a *App) onBootstrapControlPlane(s *state.State, initConfig map[string]stri
 		AllowSelfSignedCA:         true,
 		IncludeMachineAddressSANs: true,
 	})
+
+	certificates.CACert = bootstrapConfig.CACert
+	certificates.CAKey = bootstrapConfig.CAKey
+
+	certificates.FrontProxyCACert = bootstrapConfig.FrontProxyCACert
+	certificates.FrontProxyCAKey = bootstrapConfig.FrontProxyCAKey
+
+	certificates.ServiceAccountKey = bootstrapConfig.ServiceAccountKey
+
+	certificates.APIServerKubeletClientCert = bootstrapConfig.APIServerKubeletClientCert
+	certificates.APIServerKubeletClientKey = bootstrapConfig.APIServerKubeletClientKey
+
+	certificates.APIServerCert = bootstrapConfig.APIServerCert
+	certificates.APIServerKey = bootstrapConfig.APIServerKey
+	certificates.KubeletCert = bootstrapConfig.KubeletCert
+	certificates.KubeletKey = bootstrapConfig.KubeletKey
+
```

Comment:
Side note: this function is getting a bit long. I think it'd be nice to create a card to break it up a bit.

Reply:
Indeed, we should refactor the function.